### PR TITLE
i18n(hi): update outdated pages

### PR DESCRIPTION
**Pull request by at-the-vr**
#### Description

- updates missing translation for [getting-started](https://starlight.astro.build/hi/getting-started/), [components](https://starlight.astro.build/hi/guides/components/) [site-search](https://starlight.astro.build/hi/guides/site-search/) & [manual setup](https://starlight.astro.build/hi/manual-setup/) in hindi language

- related latest PRs in same order as pages linked above: #1175 #1499 #1357 #1454

### CODE REVIEW

File: docs/src/content/docs/hi/getting-started.mdx [hunk 1]
```diff
@@ -83,43 +83,9 @@ yarn dev
...
-
-```md
----
-title: नमस्ते, दुनिया !
-description: यह मेरी Starlight-संचालित साइट का एक पेज है
----
-```
-
-यदि आप कुछ भी महत्वपूर्ण भूल जाते हैं, तो Starlight आपको बता देगा।
+["पेज"](/hi/guide/pages/) गाइड में MDX और Markdoc फाइलों के लिए फ़ाइल-आधारित रूटिंग और समर्थन के बारे में और पढ़ें।
```
Comment (at-the-vr) [inline docs/src/content/docs/hi/getting-started.mdx:88]:
```suggestion
["पेज"](/hi/guides/pages/) गाइड में MDX और Markdoc फाइलों के लिए फ़ाइल-आधारित रूटिंग और समर्थन के बारे में और पढ़ें।
```

File: docs/src/content/docs/hi/guides/components.mdx [hunk 2]
```diff
@@ -195,3 +250,49 @@ import { Icon } from '@astrojs/starlight/components';
...
+<Code code={exampleCode} lang="js" title={fileName} mark={highlights} />
+```
+
+उपरोक्त कोड पेज पर निम्नलिखित उत्पन्न करता है:
+
+import { Code } from '@astrojs/starlight/components';
+
+export const exampleCode = `console.log('यह किसी फ़ाइल या CMS से आ सकता है!');`;
+export const fileName = 'example.js';
+export const highlights = ['file', 'CMS'];
```
Comment (delucis) [inline docs/src/content/docs/hi/guides/components.mdx:278]:
```suggestion
export const highlights = ['फ़ाइल', 'CMS'];
```

**Review by delucis — COMMENTED** [review]
Thank you for the updates @at-the-vr! Spotted a couple of small details but otherwise this looks great 🌟

**Review by delucis — APPROVED** [review]
Thanks for making those changes @at-the-vr — looks good to go ✨

**Pull request merged**